### PR TITLE
Achraf

### PR DESCRIPTION
**Pull request by Achraf23**
Most of coordinates_photo is complete.
The script prompts the user for the observer location and the time the photo was taken.
It then computes the longitude, latitude for a given (x,y)

For now, I put reference pixel coordinates as (x,y) in the code as show below (line 286 "coordinates_photo.py")
`    object_coordinates = get_celestial_coordinates(ref_pixel[0],ref_pixel[1],"../output/wcs_header.txt")
`
In the future, this should be done with Idriss's function that returns (x,y) values when a user click on the photo.

Hope that was clear !!

### CODE REVIEW

File: src/coordinates_photo.py [hunk 1]
```diff
@@ -111,7 +113,7 @@ def get_calibrated_camera_parameters(path: str='output/wcs_header.txt'):
             # Read the WCS solution from the file
             wcs_header_raw = f.read()
             
-            # Parse the WCS solution
+            # Parse the WCSD. Les mécanismes à l’origine d’un accident du travail
```
Comment (BloodFutur) [inline src/coordinates_photo.py:116]: WTF

File: src/coordinates_photo.py [hunk 2]
```diff
@@ -155,22 +157,138 @@ def convert_coordinates(observer_location, observer_time, object_coordinates):
...
+
+def main():
+    
+    # Prompting the user for observer location
+    observer_location_input = input("Enter observer location (latitude, longitude, altitude): ")
+    # Splitting the input string into latitude, longitude, and altitude
+    observer_location = tuple(map(float, observer_location_input.split(',')))
+
+    # Prompting the user for observer time
+    observer_time = input("Enter observer time (YYYY-MM-DD HH:MM:SS): ")
```
Comment (BloodFutur) [inline src/coordinates_photo.py:274]: Check that the input is in the right format, otherwise weird behavior might happen.

**Review by BloodFutur — CHANGES_REQUESTED** [review]
I will let you solve merge conflicts.
Did you test it on test samples?

**Comment by Achraf23** [comment]
Tout bon !

File: src/coordinates_photo.py [hunk 3]
```diff
@@ -177,38 +180,160 @@ def polar_to_cartesian(altitude, azimuth):
...
+    # Compute Greenwich Sidereal Time (GST) in degrees
+    gst = 100.46 + 0.985647 * jd
+    gst = gst % 360  # Ensure GST is within 0 to 360 degrees
+
+    # Convert GST to hours, minutes, and seconds
+    gst_hour = gst / 15
+    gst_min = (gst_hour % 1) * 60
+    gst_sec = (gst_min % 1) * 60
+
+    # Compute Local Sidereal Time (LST) in degrees
```
Comment (annzzza) [inline src/coordinates_photo.py:199]: time conversion is much clearer now, thanks!

File: src/coordinates_photo.py [hunk 4]
```diff
@@ -177,38 +180,160 @@ def polar_to_cartesian(altitude, azimuth):
...
+
+def cartesian_to_lon_lat(x,y):
+    """
+    Submethod to convert cartesian coordinates (x,y,z) (we assume z = 10km) 
+    to longitude, latitude in radians
+    Returns
+    -------
+    (lon, lat) : tuple
+    """
+    z = 10000 # en m
```
Comment (annzzza) [inline src/coordinates_photo.py:265]: if possible make z an optional parameter for this function, with default value 10km

File: src/coordinates_photo.py [hunk 5]
```diff
@@ -177,38 +180,160 @@ def polar_to_cartesian(altitude, azimuth):
...
+
+    # Splitting the input string into latitude, longitude, and altitude
+    observer_location = tuple(map(float, observer_location_input.split(',')))
+
+    
+    # Define the expected format
+    expected_format = r"\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}"
+
+    # Prompting the user for observer time
+    observer_time = input("Enter observer time (YYYY-MM-DD HH:MM:SS): ")
```
Comment (annzzza) [inline src/coordinates_photo.py:312]: I imagine the time should be given with respects to greenwich mean time ? If so, make it clear to the user when requesting input date and time.

File: src/coordinates_photo.py [hunk 6]
```diff
@@ -177,38 +180,160 @@ def polar_to_cartesian(altitude, azimuth):
...
+    
+    # Prompting the user for observer location
+    observer_location_input = input("Enter observer location (latitude, longitude, altitude): ")
+
+    # Check if input matches the expected format
+    if re.match(expected_format, observer_location_input):
+        print("Input is in the correct format.")
+    else:
+        print("Input is not in the correct format. Please enter in latitude, longitude, altitude format.")
+        return -1
```
Comment (annzzza) [inline src/coordinates_photo.py:302]: <optional> Instead of returning, shouldn't we loop until the input fits the format?

**Review by annzzza — CHANGES_REQUESTED** [review]
Let us know if you tried the program on the 'testing data' provided in the google drive

File: src/coordinates_photo.py (hunk 5 above)
Comment (Achraf23) [reply src/coordinates_photo.py:312]: No need. Time is converted in greenwich inside 'convert_coordinates' function. Below is the code
`observer_time = Time(observer_time)
    # Compute Julian Date (JD)
    jd = observer_time.jd

    # Compute Greenwich Sidereal Time (GST) in degrees
    gst = 100.46 + 0.985647 * jd
    gst = gst % 360  # Ensure GST is within 0 to 360 degrees`

**Review by elouali-code — COMMENTED** [review]
For me I think it's all good

File: src/coordinates_photo.py [hunk 7]
```diff
@@ -134,81 +139,145 @@ def get_calibrated_camera_parameters(path: str='output/wcs_header.txt'):
...
+    # Calculate intermediate angle
+    psi_alt = np.pi/2 - elevation_rad - np.arcsin(Earth_radius/(Earth_radius+aircraft_alt)*np.cos(elevation_rad))
+    
+    #calculate lat
+    lat = np.arcsin(np.sin(observer_lat_rad) * np.cos(psi_alt) + 
+                        np.cos(observer_lat_rad) * np.sin(psi_alt) * np.cos(azimuth_rad))
     
-    azimuth, altitude = equatorial_to_azimuthal(object_ra, object_dec, lst, observer_lat)
+    # Calclute longitude based on latitude's val
+    if (observer_lat>70   and  np.tan(psi_alt)*np.cos(azimuth_rad) > np.tan(np.pi/2-observer_lat_rad)) or \
```
Comment (annzzza) [inline src/coordinates_photo.py:205]: source for formulas?

**Review by annzzza — APPROVED** [review]
give information about your teste: did they pass? how precise are you?

**Pull request merged**